### PR TITLE
Libmesh update

### PR DESCRIPTION
**Pull request by roystgnr**
- Even more internal smart pointers.  This includes changes to the `Parameter` class that may require downstream changes to match; compatibility in Moose itself has already been fixed.
- When trying to combine a low-order `Elem` with an incompatibly higher order `FE` type, the mistake is now always reported with
  comprehensible error message, even in non-debugging-mode builds.
- When a parsed function is used and the parser returns an error, we now check and report details even in non-debugging-mode builds.
- The last use of C++17-deprecated std::iterator has been fixed; a few compilers outside CI were giving us warnings about it.
- We put support for input meshes that themselves have holes into MeshedHole, so you can mesh with holes while your hole has holes.
- Bug fixes for some triangulator use cases.
- Compatibility fixes for some `PETSc` 3.17 configurations.

- MetaPhysicL updates, for evaluating `pow(negative DualNumber, integer)` operations without throwing floating point exceptions, and also evaluating binary functions faster in general when mixing DualNumber and non-DualNumber parameter types.

I mostly need this for the triangulator fixes, but the MetaPhysicL updates fix a Moose-user-reported bug.  I also want to get that `Parameters` change into Moose CI, to be completely sure it doesn't affect anything that we didn't previously catch via libMesh CI.

### CODE REVIEW

**Comment by roystgnr** [comment]
Hooray for getting things into CI early.

I saw the red X, was eagerly deciding how to castigate John for being too insistent on pushing that PR without more work on backwards compatibility ... but the breakage here has absolutely nothing to do with his `Parameters` improvements; I've regressed something in MetaPhysicL.  Looks like it won't be hard to replicate upstream and fix, at least.

**Review by loganharbour — CHANGES_REQUESTED** [review]
Make sure to mix the submodule update + mpich yaml file change into the same commit, and put it last

File: modules/doc/content/newsletter/2022_05.md [hunk 1]
```diff
@@ -45,6 +45,29 @@ for a complete description of all MOOSE changes.
...
   support for `std::isinf()`.
 
+### `2022.05.24` Update
+
+- Even more internal smart pointers.  This includes changes to the
+  `Parameters` class that may require downstream changes to match;
+  compatibility in Moose itself has already been fixed.
+- When trying to combine a low-order `Elem` with an incompatibly
+  higher order `FE` type, the mistake is now always reported with
+  comprehensible error message, even in non-debugging-mode builds.
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2022_05.md:55]:
```suggestion
  higher order `FE` type, the mistake is now always reported with a
  comprehensible error message, even in non-debug-mode builds.
```

File: modules/doc/content/newsletter/2022_05.md [hunk 2]
```diff
@@ -45,6 +45,29 @@ for a complete description of all MOOSE changes.
...
+- When a parsed function is used and the parser returns an error, we
+  now check and report details even in non-debugging-mode builds.
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2022_05.md:57]:
```suggestion
  now check and report details even in non-debug-mode builds.
```

File: modules/doc/content/newsletter/2022_05.md [hunk 3]
```diff
@@ -45,6 +45,29 @@ for a complete description of all MOOSE changes.
...
+- The last use of C++17-deprecated std::iterator has been fixed;
+  a few compilers outside CI were giving us warnings about it.
+- We put support for input meshes that themselves have holes into
+  MeshedHole, so you can mesh with holes while your hole has holes.
+- Bug fixes for some triangulator use cases.
+- Compatibility fixes for some `PETSc` 3.17 configurations.
+
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2022_05.md:64]:
```suggestion
```

File: modules/doc/content/newsletter/2022_05.md [hunk 4]
```diff
@@ -45,6 +45,29 @@ for a complete description of all MOOSE changes.
...
+- MetaPhysicL updates, for evaluating `pow(negative DualNumber,
+  integer)` operations without throwing floating point exceptions,
+  and also evaluating binary functions faster in general when mixing
+  DualNumber and non-DualNumber parameter types.
+
+
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2022_05.md:70]:
```suggestion
```

File: modules/doc/content/newsletter/2022_05.md [hunk 5]
```diff
@@ -45,6 +45,29 @@ for a complete description of all MOOSE changes.
...
+  compatibility in Moose itself has already been fixed.
+- When trying to combine a low-order `Elem` with an incompatibly
+  higher order `FE` type, the mistake is now always reported with
+  comprehensible error message, even in non-debugging-mode builds.
+- When a parsed function is used and the parser returns an error, we
+  now check and report details even in non-debugging-mode builds.
+- The last use of C++17-deprecated std::iterator has been fixed;
+  a few compilers outside CI were giving us warnings about it.
+- We put support for input meshes that themselves have holes into
+  MeshedHole, so you can mesh with holes while your hole has holes.
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2022_05.md:61]:
```suggestion
  `MeshedHole`, so you can mesh with holes while your hole has holes.
```

**Review by cticenhour — CHANGES_REQUESTED** [review]
A few news comments

**Review by loganharbour — DISMISSED** [review]
While we're at it, let's add `--enable-xdr-required` in `configure_libmesh.sh`.

**Comment by roystgnr** [comment]
> loganharbour force-pushed the libmesh_update_202205 branch from af82f58 to e00c192 5 hours ago

Wow - what just happened to get us passing the "Min clang" check!?  My workaround in MetaPhysicL was really all it took?  That's fantastic.

**Comment by loganharbour** [comment]
Magic!

Just did one more push; I forgot to update the build string.

**Review by milljm — APPROVED** [review]
👍

**Review by cticenhour — APPROVED** [review]
I will move this news update to the proper spot in June as part of my news PR. Otherwise looks good.

**Pull request merged**